packages/docusaurus-theme-classic/src/theme/DocCardList/index.tsx

```diff
@@ -9,6 +9,17 @@ import React from 'react';
 
 import DocCard from '@theme/DocCard';
 import type {PropSidebarItem} from '@docusaurus/plugin-content-docs';
+import {findFirstCategoryLink} from '@docusaurus/theme-common';
+
+// Filter categories that don't have a link.
+function filterItems(items: PropSidebarItem[]): PropSidebarItem[] {
+  return items.filter((item) => {
+    if (item.type === 'category') {
+      return !!findFirstCategoryLink(item);
```

Comment:
That seems more expensive than before because we traverse the entire subtree twice. Is it worthwhile?

Reply:
Oh, otherwise we render an empty `article` container😅 Well, fair then. A little nasty

Reply:
> That seems more expensive than before

Agree but at the same time it's unlikely that the user build complex sidebar trees without any link, in most cases the link is found fast

Reply:
Yeah, makes sense 👍